int32 velA
int32 velB
uint8 laser
